### ISSUE TITLE
Create shortcuts that call local programs

### ISSUE THREAD

**Issue by Spaghetti-Cat** [Lord-Kamina/keybinder#14]
It would be make this addon extremely extensible and useful if we could create binds that call, for example, shell scripts. My use case is when I'm viewing images online I could download them to separate folders thereby organising them (The normal way of "Right Click -> Save Image -> Browse to directory -> Click save" is too cumbersome if you save more than a handful of images.

If a URL could be passed as an argument the possibilities are limitless! This feature was available in Opera, but in the years since I switched to Firefox I could not find an alternative.

PS: Thank you for this nice plugin :)

**Comment by Lord-Kamina** [comment]
Hmm that might be theoretically possible, I haven't actually checked into
it, and I don't know if I'd still be able to host the addon on AMO with a
feature like that. Not exactly what you want but have you tried some of the
image saving add-ons out there?

**Comment by Spaghetti-Cat** [comment]
Huh, well.. I didn't think of getting a dedicated addon just to download a picture. But, by God! With [Save Images in Folder](https://addons.mozilla.org/en-US/firefox/addon/save-image-in-folder/) I was able to add download locations to the context menu and those menu items would download the images when clicked. Then I used [Menu Wizard](https://addons.mozilla.org/en-US/firefox/addon/s3menu-wizard/) to move those menu items into the root context menu for much faster access.

Thank you! For a year now I was saving images by the hunderds the old way, and you led me to the easy solution.

I will use this method, but it would still be cool to have keyboard shortcuts that call shell scripts or even other programs, e.g. download the passed URL and open it in an image ediitng program. It gives A LOT more control.
